cerberus/client.py

```diff
@@ -668,3 +668,28 @@ def secret_merge(self, secure_data_path, key):
         # unicode stings as the payload.
         combined_key = json.encoder.JSONEncoder().encode(temp_key)
         return combined_key
+
+    def get_metadata(self, sdb_name=None):
+        """
+        (Admin) get metadata for all SDBs or a specific one when sdb_name is set
+        Keyword arguments:
+            sdb_name (string) -- The name of the SDB to query for
+        """
+        metadata_list = []
+
+        metadata_resp = self._get_metadata_resp(sdb_name=sdb_name)
+        metadata_list.extend(metadata_resp.get('safe_deposit_box_metadata'))
+        while metadata_resp['has_next']:
```

Comment:
In another implementation of dealing with offset, we set `has_next` and `next_offset`. Is that necessary for this case?

https://github.com/Nike-Inc/cerberus-python-client/blob/master/cerberus/client.py#L431

Reply:
It's another way to make the while loop guaranteed to execute once since do..while is not a thing in Python. Either way there's going to be some extra logic. In my case I decided to explicitly execute identical content before and during the while loop. I can also take out the lines before the while loop and replace them with the above lines.